nautobot/utilities/forms/widgets.py

```diff
@@ -135,7 +135,7 @@ def __init__(self, api_url=None, full=False, *args, **kwargs):
 
         self.attrs["class"] = "nautobot-select2-api"
         if api_url:
-            self.attrs["data-url"] = "/{}{}".format(settings.BASE_PATH, api_url.lstrip("/"))  # Inject BASE_PATH
+            self.attrs["data-url"] = api_url
```

Comment:
FYI this is the only one that is potentially problematic, and instead of trying to hard-code the URL here, we should address all the places where the API URL is being hard-coded in calling code so that they can also naturally inherit the correctly namespaced URLs.